apps/dashboard/src/components/LeftPanel/ElementRow.tsx

```diff
@@ -29,11 +32,47 @@ export function ElementRow({ element }: ElementRowProps) {
     transition,
   };
 
+  const [isEditing, setIsEditing] = useState(false)
+  const formRef = useRef<HTMLFormElement>(null)
+
+  function onSubmit(event: FormEvent) {
+    event.preventDefault()
+
+    if (!element.name) {
+      let defaultName = ""
+
+      if (element.tag === 'p') {
+        defaultName = "Text"
+      } else if (element.tag === 'div' && element.backgroundImage) {
+        defaultName = "Image"
+      } else if (element.tag === 'div' && !element.backgroundImage && !element.radius) {
+        defaultName = "Box"
+      } else if (element.tag === 'div' && !element.backgroundImage && element.radius) {
+        defaultName = "Rounded box"
+      } else {
+        defaultName = "Dynamic text"
+      }
+
+      updateElement({
+        ...element,
+        name: defaultName
+      })
+    }
+
+    setIsEditing(false)
+  }
+
   return (
     <div className="flex justify-between items-center pb-2 cursor-auto" ref={setNodeRef} style={style} {...attributes} {...listeners}>
       <button
-        className={`flex gap-2 select-none text-gray-600 hover:text-gray-900 w-full ${selectedElement === element.id ? '!text-blue-500' : ''} ${!element.visible ? '!text-gray-300' : ''}`}
+        className={`flex items-center gap-2 select-none text-gray-600 hover:text-gray-900 w-full ${selectedElement === element.id ? '!text-blue-500' : ''} ${!element.visible ? '!text-gray-300' : ''}`}
         onClick={() => { setSelectedElement(element.id); }}
+        onDoubleClick={() => {
+          if (isEditing)
+            return
+
+          setIsEditing(true)
```

Comment:
There seem to be an issue where we don't focus the input right after double-clicking, we means we have to double-click + click again to actually start editing the name:

https://github.com/QuiiBz/ogstudio/assets/43268759/784e4ed4-969f-4378-b715-0d4ca9bc3fd9

Reply:
I think setting autofocus on inputs will solve the issue.